### PR TITLE
Add manifest libraries

### PR DESCRIPTION
**Pull request by liamawhite**
Signed-off-by: Liam White <liam@tetrate.io>

### CODE REVIEW

File: pkg/manifest/fetch.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+)
+
+// Fetch retrieves and parses a manifest from the URL passed
+func Fetch(manifestURL string) (*api.Manifest, error) {
+	resp, err := http.Get(manifestURL)
+	if err != nil {
+		return nil, err
+	}
+	if resp.StatusCode > 200 {
+		return nil, fmt.Errorf("received %q from %v", resp.StatusCode, manifestURL)
```
Comment (taiki45) [inline pkg/manifest/fetch.go]: nit: Logging the error response in a debug level might help a debugging. Not necessary this time though.

File: pkg/manifest/fetch_test.go [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+	return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(responseStatusCode)
+		if responseStatusCode == 200 {
+			bytes, _ := ioutil.ReadFile(filepath.Join("testdata", responseManifestFile))
+			w.Write(bytes)
+		}
+	}))
+}
+
+func goodManifest() *api.Manifest {
```
[the remark this answers is not in the document]
Comment (liamawhite) [reply pkg/manifest/fetch_test.go]: We could eventually, util packages, in general, are frowned upon in go though.

File: pkg/manifest/testdata/manifest.golden [hunk 3]
```diff
@@ -0,0 +1,102 @@
+{
+  "version": "v0.1.0",
```
[the remark this answers is not in the document]
Comment (liamawhite) [reply pkg/manifest/testdata/manifest.golden]: this sounds more sensible will change to `manifest_version`.

File: pkg/manifest/fetch.go [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+	"github.com/tetratelabs/getenvoy/api"
+)
+
+// Fetch retrieves and parses a manifest from the URL passed
+func Fetch(manifestURL string) (*api.Manifest, error) {
+	resp, err := http.Get(manifestURL)
+	if err != nil {
+		return nil, err
+	}
+	if resp.StatusCode >= 400 {
```
Comment (taiki45) [inline pkg/manifest/fetch.go]: We should have more strict status code spec here like only accept 200 as a successful results.

Comment (liamawhite) [reply pkg/manifest/fetch.go]: That sounds reasonable, we can always change it.

**Pull request merged**